core/trino-main/src/main/java/io/trino/operator/aggregation/partial/PartialAggregationController.java

```diff
@@ -13,12 +13,15 @@
  */
 package io.trino.operator.aggregation.partial;
 
+import io.airlift.units.DataSize;
 import io.trino.operator.HashAggregationOperator;
 
+import static java.util.Objects.requireNonNull;
+
 /**
  * Controls whenever partial aggregation is enabled across all {@link HashAggregationOperator}s
  * for a particular plan node on a single node.
- * Partial aggregation is disabled once enough rows has been processed ({@link #minNumberOfRowsProcessed})
+ * Partial aggregation is disabled once enough rows has been processed)
```

Comment:
once enough rows has been processed -> after sampling sufficient amount of input